src/Microsoft.Extensions.Diagnostics.HealthChecks/CompositeHealthCheckResult.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.Extensions.Diagnostics.HealthChecks
+{
+    /// <summary>
+    /// Represents the results of multiple health checks.
+    /// </summary>
+    public class CompositeHealthCheckResult
+    {
+        /// <summary>
+        /// A <see cref="IReadOnlyDictionary{TKey, T}"/> containing the results from each health check.
+        /// </summary>
+        /// <remarks>
+        /// The keys in this dictionary map to the name of the health check, the values are the <see cref="HealthCheckResult"/>
+        /// returned when <see cref="IHealthCheck.CheckStatusAsync(System.Threading.CancellationToken)"/> was called for that health check.
+        /// </remarks>
+        public IReadOnlyDictionary<string, HealthCheckResult> Results { get; }
+
+        /// <summary>
+        /// Gets a <see cref="HealthCheckStatus"/> representing the aggregate status of all the health checks.
+        /// </summary>
+        /// <remarks>
+        /// This value is determined by taking the "worst" result of all the results. So if any result is <see cref="HealthCheckStatus.Failed"/>,
+        /// this value is <see cref="HealthCheckStatus.Failed"/>. If no result is <see cref="HealthCheckStatus.Failed"/> but any result is
+        /// <see cref="HealthCheckStatus.Unhealthy"/>, this value is <see cref="HealthCheckStatus.Unhealthy"/>, etc.
+        /// </remarks>
+        public HealthCheckStatus Status { get; }
+
+        /// <summary>
+        /// Create a new <see cref="CompositeHealthCheckResult"/> from the specified results.
+        /// </summary>
+        /// <param name="results">A <see cref="IReadOnlyDictionary{TKey, T}"/> containing the results from each health check.</param>
+        public CompositeHealthCheckResult(IReadOnlyDictionary<string, HealthCheckResult> results)
+        {
+            Results = results;
+            Status = CalculateAggregateStatus(results.Values);
+        }
+
+        private HealthCheckStatus CalculateAggregateStatus(IEnumerable<HealthCheckResult> results)
+        {
+            // This is basically a Min() check, but we know the possible range, so we don't need to walk the whole list
+            var currentValue = HealthCheckStatus.Healthy;
```

Comment:
Should this start off as `Unknown` (in the event no results are configured)?

Reply:
If no checks are configured, the result should be `Healthy`

Reply:
Sometimes you don't have anything to check other than "can you ping the service" its often enough that you get a response.